File: lecture_010(RecursionwithArraylists)/l_008_printstairspath.cpp
#include<iostream>
#include<vector>
#include<string>
using namespace std;

void printstairspath(int n, string path) {
    if(n==0) {
        cout<<path<<endl;
        return;
    }
    else if(n<0) {
        return;
    }
    printstairspath(n-1,path+"1");
    printstairspath(n-2,path+"2");
    printstairspath(n-3,path+"3");
}

int main() {
    int n;
    cin>>n;
    printstairspath(n,"");

    return 0;
}

File: lecture_010(RecursionwithArraylists)/l_008_printstairspath.java
import java.util.Scanner;
public class l_008_printstairspath {
	public static Scanner scn = new Scanner(System.in);
	public static void main(String[] args) {
		int n = scn.nextInt();
		printStairsPath(n,"");
	}
	public static void printStairsPath(int n, String path) {
		if(n==0) {
			System.out.println(path);
		}
		else if(n<0) {
			return;
		}
		printStairsPath(n-1,path+"1");
		printStairsPath(n-2,path+"2");
		printStairsPath(n-3,path+"3");
	}
}